wtf_tinymce/forms/fields.py

```diff
@@ -107,7 +106,7 @@ def process_formdata(self, value_list):
                 attributes=self.sanitize_attributes
             )
 
-        if self.linkify:
+        if self.linkify and self.data:
```

Comment:
In which cases self.data can be None here?

Reply:
I think I got it during unit tests.
Will try to run the tests again without the check.

Reply:
In wtforms/fields/core.py, process_formdata is called with self.raw_data, which can be an empty list if there is something in the form (formdata), but not the field we need:

        if formdata is not None:
            if self.name in formdata:
                self.raw_data = formdata.getlist(self.name)
            else:
                self.raw_data = []

            try:
                self.process_formdata(self.raw_data)
            except ValueError as e:
                self.process_errors.append(e.args[0])
                
At this point, self.data is None and the error is triggered.

I'm having a hard time creating a simple test for this, but if needed I can look deeper into it.

Reply:
from flask_testing import TestCase
from flask_wtf import FlaskForm
from webapp.wtf_tinymce.forms.fields import TinyMceField
from flask import Flask


class TestWtfTinymce(TestCase):

    def create_app(self):
        app = Flask(__name__)
        app.config['TESTING'] = True
        app.config['SECRET_KEY'] = 'secret'
        return app

    def test_raises(self):
        class TestForm(FlaskForm):
            a = TinyMceField('a')
            b = TinyMceField('b')

        with self.app.test_request_context(path='/', method='POST', data={'a': 'sdfsdf', 'b': 'foo'}):
            TestForm()

        with self.app.test_request_context(path='/', method='POST', data={'a': 'sdfsdf'}):
            TestForm()

Reply:
Thank you very much for detailed explanation and problem solving.